### ISSUE TITLE
Slice requires two parameters

### ISSUE THREAD

**Issue by rogerl50** [ArnoldSmith86/virtualtabletop#1084]
`"var foo = 'abcde' slice 1"` returns the empty string. One must write `"var foo = 'abcde' slice 1 5"` to get the last four characters. This is contrary to how `slice` works in Javascript, and seems to contradict the documentation in the JSON editor as well.

**Comment by robartsd** [comment]
I think the second parameter is getting called as null or an empty string, but being tested for undefined. `"abcde".slice(1, null)` and `"abcde".slice(1,"")` both return undefined which is probably being converted to empty string.

**Comment by robartsd** [comment]
Other compute functions that test for `undefined`:
- join
- hypot
- charAt
- charCodeAt
- codePointAt
- match
- toFixed
- toLocaleLowercase
- toLocaleUppercase
- substr

**Comment by rogerl50** [comment]
`"var x = 'abcde' slice 1"` shows the following in the debugger:
<img width="737" alt="Screen Shot 2022-01-24 at 12 59 52 PM" src="https://user-images.githubusercontent.com/63379656/150838527-f8469523-f1c1-421b-b66c-d4c10744e6a3.png">
So it looks like the second (empty) parameter is being converted to 1, probably by the variable evaluation code.

**Comment by rogerl50** [comment]
And indeed this code:
```
const getValue = function(input) {
            const toNum = s=>typeof s == 'string' && s.match(/^[-+]?[0-9]+(\.[0-9]+)?$/) ? +s : s;
            if(match[14] && match[9] !== undefined)
              return compute(match[13] ? variables[match[14]] : match[14], input, toNum(getParam(9, 1)), toNum(getParam(15, 1)), toNum(getParam(19, 1)));
            else if(match[14])
              return compute(match[13] ? variables[match[14]] : match[14], input, toNum(getParam(15, 1)), toNum(getParam(19, 1)), toNum(getParam(23, 1)));
            else
              return getParam(5, null);
          };
```
in widgets.js is converting undefined parameters to 1.

**Comment by robartsd** [comment]
That's right, you can pass explicit null values, but undefined values default to one; however, slice does not work with null.

**Comment by rogerl50** [comment]
So are you suggesting that we close this and change the documentation to require the second parameter on slice as well as the other functions you listed above?
